pages/v3-migration.page.tsx

```diff
@@ -0,0 +1,151 @@
+import { Trans } from '@lingui/macro';
+import { Box, Divider } from '@mui/material';
+import { useEffect } from 'react';
+import { ConnectWalletPaper } from 'src/components/ConnectWalletPaper';
+import { ContentContainer } from 'src/components/ContentContainer';
+import { MigrateV3Modal } from 'src/components/transactions/MigrateV3/MigrateV3Modal';
+import { useAppDataContext } from 'src/hooks/app-data-provider/useAppDataProvider';
+import { usePermissions } from 'src/hooks/usePermissions';
+import { useUserReserves } from 'src/hooks/useUserReserves';
+import { MainLayout } from 'src/layouts/MainLayout';
+import { useWeb3Context } from 'src/libs/hooks/useWeb3Context';
+import { DashboardContentNoData } from 'src/modules/dashboard/DashboardContentNoData';
+import { MigrationBottomPanel } from 'src/modules/migration/MigrationBottomPanel';
+import { MigrationListItem } from 'src/modules/migration/MigrationListItem';
+import { MigrationListItemLoader } from 'src/modules/migration/MigrationListItemLoader';
+import { MigrationLists } from 'src/modules/migration/MigrationLists';
+import { MigrationTopPanel } from 'src/modules/migration/MigrationTopPanel';
+import { usePoolDataV3Subscription, useRootStore } from 'src/store/root';
+
+export default function V3Migration() {
+  const { loading } = useAppDataContext();
+  const { currentAccount, loading: web3Loading } = useWeb3Context();
+  const { isPermissionsLoading } = usePermissions();
+  const setCurrentMarketForMigration = useRootStore((state) => state.setCurrentMarketForMigration);
+
+  const { user, borrowPositions } = useUserReserves();
+
+  const {
```

Comment:
The benefits of putting selected assets in global state are
- trigger selectors changes on selecting any asset (HF calculation)
- construction permits and approvals  without passing anything to modal
- select appropriate state for migration right from the store instead of passing to migrate method and mapping there
Benefits of local state
- No "hanging" state

Since I was using selectors heavily, that's why I decided to keep selected assets in global state instead of a local one

Reply:
Yeah, good point, I'm not a huge fan of local scope as it mixes lifecycle everywhere and nested simple selector functions is a really handy feature when you need to map a lot data in slightly different way, for example increasing amount for migrate with borrow, which should be exactly the same for both permits payloads and passing exact same data to .migrate method from utilities. With global state it's just less error prone, as you simply can't pass wrong data. 
But consistency is very important, so I agree, just wanted to explain my initial point

Reply:
Thinking about it again, I think even with the rest of the app using different approach, we need to consider using global state instead of local one and this particular case could be a good example. 
The bulk of the problem with transaction flow comes due to local state being coupled with hooks and instead of directly accessing methods and state it we need to pass it above using some hook "factory" while both passing props to children **and** using global context. 
Which introduce it's own set of challenges as now it's harder to mix multiple actions together, because the state is reactive
Do 1 wait, do 2 wait, set the state is becoming non-trivial, that's why in the code there are setTimeouts to handle async states. 

With this particular example we never need to worry if the state passed to migrate function is the latest state, if the passed arguments are passed from the proper parent. 
There is literally 1 single entry point which is impossible to miss and all the derived state will always be up to date regardless if we use for methods, derrived state or anything else. 
I'm not saying it is the best approach, we might find better one, but being holistic will not help, as the current approach is both complicated to understand, custom and non-flexible. 

It's not related to wrong chain id and using ActionModals for migration the same way, that is a good thing. But we need to think how to reduce multiple places holding equally important business logic and state. 

Would love to discuss pros and cons